Dockerfile

```diff
@@ -1,31 +1,110 @@
-FROM ubuntu:xenial
+FROM ubuntu:xenial as base
 
 RUN apt-get update && apt-get -y install \
-    git \
-    net-tools \
     sudo \
-    wget \
-&& rm -rf /var/lib/apt/lists/*
-
-RUN useradd --create-home -s /bin/bash ubuntu
-RUN adduser ubuntu sudo
-RUN echo -n 'ubuntu:ubuntu' | chpasswd
+    curl \
+    locales \
+    && rm -rf /var/lib/apt/lists/*
+RUN locale-gen en_US.UTF-8
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8
 
-# Enable passwordless sudo for users under the "sudo" group
-RUN sed -i.bkp -e \
+# Add fixuid to change permissions for bind-mounts. Set uid to same as host with -u <uid>:<guid>
+RUN addgroup --gid 1000 docker && \
+    adduser --uid 1000 --ingroup docker --home /home/docker --shell /bin/sh --disabled-password --gecos "" docker && \
+    usermod -aG sudo docker && \
+    sed -i.bkp -e \
       's/%sudo\s\+ALL=(ALL\(:ALL\)\?)\s\+ALL/%sudo ALL=NOPASSWD:ALL/g' \
       /etc/sudoers
+RUN USER=docker && \
+    GROUP=docker && \
+    curl -SsL https://github.com/boxboat/fixuid/releases/download/v0.3/fixuid-0.3-linux-amd64.tar.gz | tar -C /usr/local/bin -xzf - && \
+    chown root:root /usr/local/bin/fixuid && \
+    chmod 4755 /usr/local/bin/fixuid && \
+    mkdir -p /etc/fixuid && \
+    printf "user: $USER\ngroup: $GROUP\npaths:\n  - /service\n" > /etc/fixuid/config.yml
 
-USER ubuntu
+RUN echo "LANG=C.UTF-8" > /etc/default/locale
 
-ADD . /home/ubuntu/IncludeOS
-WORKDIR /home/ubuntu/IncludeOS
+# Docker TAG can be specified when building with --build-arg TAG=..., this is redeclared in the source-build stage
+# Git tags can be specified with --build-arg NEWTAG=..., the default is set as includeos-dev
+ARG BRANCH=dev
+ARG REPO=hioa-cs
+ARG NEWTAG=includeos-dev
+ENV BRANCH=$BRANCH
+ENV REPO=$REPO
+ENV NEWTAG=$NEWTAG
 
-RUN sudo apt-get update && \
-    sudo do_bridge="" ./etc/install_all_source.sh \
-&& sudo rm -rf /var/lib/apt/lists/*
 
-VOLUME /service
+LABEL dockerfile.version=1 \
+      includeos.version=$BRANCH
 WORKDIR /service
 
-CMD ./run.sh
+#########################
+FROM base as source-build
+
+RUN apt-get update && apt-get -y install \
+    git \
+    lsb-release \
+    net-tools \
+    wget \
+    && rm -rf /var/lib/apt/lists/*
+
+
+# Copy IncludeOS contents from host repo to container
+RUN echo "copying Contents of current IncludeOS branch"
+RUN cd ~ && pwd && \
+  mkdir -p IncludeOS
+COPY . /root/IncludeOS/
+
+# Adding Custom git tags
+RUN echo "Assigning Your custom git tag"
```

Comment:
Either print what you sent in as custom tag, or remove the print.
Also there should not be a default tag. If no tag was sent in then it should just keep the same tag as was there from before. (make sure this works even if we are on the latest tag)